rs/src/management/http_client.rs

```diff
@@ -84,11 +100,16 @@ impl TunnelManagementClient {
     /// Creates a new tunnel.
     pub async fn create_tunnel(
         &self,
-        tunnel: &Tunnel,
+        mut tunnel: Tunnel,
         options: &TunnelRequestOptions,
     ) -> HttpResult<Tunnel> {
-        let url = self.build_uri(tunnel.cluster_id.as_deref(), TUNNELS_API_PATH);
-        let mut request = self.make_tunnel_request(Method::POST, url, options).await?;
+        if tunnel.tunnel_id.is_none() || tunnel.tunnel_id.as_ref().unwrap().is_empty() {
+            tunnel.tunnel_id = Some(TunnelManagementClient::generate_tunnel_id());
+        }
+        let mut url = self.build_uri(tunnel.cluster_id.as_deref(), TUNNELS_API_PATH);
+        let new_path = url.path().to_owned() + "/" + tunnel.tunnel_id.as_ref().unwrap();
+        url.set_path(&new_path);
+        let mut request = self.make_tunnel_request(Method::PUT, url, options).await?;
```

Comment:
I would go with something like this to avoid the extra unwraping

```suggestion
        let mut url = self.build_uri(tunnel.cluster_id.as_deref(), TUNNELS_API_PATH);
        let tunnel_id = match tunnel.tunnel_id.as_ref() {
            None | Some("") => TunnelManagementClient::generate_tunnel_id(),
            Some(tunnel_id) => tunnel_id.to_owned(),
        };
        let new_path = url.path().to_owned() + "/" + tunnel_id;
        url.set_path(&new_path);
        let mut request = self.make_tunnel_request(Method::PUT, url, options).await?;
```

Reply:
I fixed it to a working configuration, the Some("") was causing an error in your suggestion 
![image](https://github.com/microsoft/dev-tunnels/assets/12804248/7a7cc627-7ff3-486c-a53f-b3d5df143b3d)